discovery-provider/src/api/v1/users.py

```diff
@@ -947,3 +954,74 @@ def get(self, id: str):
         support = get_support_sent_by_user(args)
         support = list(map(extend_supporting, support))
         return success_response(support)
+
+
+verify_token_response = make_response(
+    "verify_token", ns, fields.List(fields.Nested(decoded_user_token))
+)
+
+
+@ns.route("/verify_token")
+class GetTokenVerification(Resource):
+    @record_metrics
+    @ns.doc(
+        id="""Verify ID token""",
+        description="""Verify if the given jwt ID token was signed by the subject (user) in the payload""",
+        responses={
+            200: "Success",
+            400: "Bad input",
+            404: "ID token not valid",
+            500: "Server error",
+        },
+    )
+    @ns.expect(verify_token_parser)
+    @ns.marshal_with(verify_token_response)
+    def get(self):
+        args = verify_token_parser.parse_args()
+        # 1. Break JWT into parts
+        raw_token = args["token"]
+        if not raw_token:
+            abort_bad_request_param("token", ns)
+        token_parts = args["token"].split(".")
+        if not len(token_parts) == 3:
+            abort_bad_request_param("token", ns)
+
+        # 2. Decode the signature
+        try:
+            signature = base64.urlsafe_b64decode(token_parts[2] + "==")
```

Comment:
Is there a reason to add the "==" here instead of including it from the client?

Reply:
Yeah this is just padding to make sure we don't run into any decoding issues, since this function throws if the length is not divisible by 4. Padding is optional for urlsafe-base64 so I left it out in the JWT which seems to be convention